resin/settings.py

```diff
@@ -35,15 +36,20 @@ class Settings(object):
                                             'cache')
 
     def __init__(self):
-        if Path.isdir(self._setting['data_directory']):
+        try:
+            self.__read_settings()
+        except:
+            # move old settings file if it exists.
             if Path.isfile(Path.join(self._setting['data_directory'],
                                      self.CONFIG_FILENAME)):
-                self.__read_settings()
-            else:
-                self.__write_settings()
-        else:
-            # data directory doesn't exist
+                shutil.move(Path.join(self._setting['data_directory'],
+                                     self.CONFIG_FILENAME)
+                            , Path.join(self._setting['data_directory'],
+                                     "{0}.{1}".format(self.CONFIG_FILENAME,'old')))
             self.__write_settings()
+            raise exceptions.InvalidSettings(Path.join(
+                                            self._setting['data_directory'],
+                                            self.CONFIG_FILENAME))
```

Comment:
@dfunckt I don't think we should do that while we couldn't correct the settings file. User might have some custom settings and we will go wrong if we write default settings and keep the app running.